### PR TITLE
Add py.typed file recommended by PEP 561 (v1.x)

### PR DESCRIPTION
**Pull request by antoinebrl**
Follow recommendations from [PEP 561 – Distributing and Packaging Type Information](https://peps.python.org/pep-0561/)

### CODE REVIEW

**Comment by SiddhantSadangi** [comment]
Hey @antoinebrl 👋 
Thanks for this PR, and I'm sorry for the delay in getting back to you (we were at ICML last week).

Could you please point this PR to [neptune-ai:dev/1.x](https://github.com/neptune-ai/neptune-client/tree/dev/1.x)?

**Comment by antoinebrl** [comment]
@SiddhantSadangi, I rebased and changed the destination. I opened a similar PR to target v2.0 https://github.com/neptune-ai/neptune-client/pull/1855

**Comment by SiddhantSadangi** [comment]
Hey @antoinebrl,
After further internal discussions, considering the focus on the 2.x client, we've decided not to invest effort into improving type-hinting in 1.x.
I am closing this PR as not planned, but keeping #1855 open.

**Comment by antoinebrl** [comment]
Does this mean there will be no more releases in v1? If that isn't true or certain, I would still be in favor or adding this empty file to be compliant with PEP 561.

**Comment by antoinebrl** [comment]
Hello! As I suspected, there was another release in the v1 series. Unfortunately, by closing the PR, the package is still considered untyped.

**Comment by SiddhantSadangi** [comment]
Hey @antoinebrl , we might have a bugfix release in the v1 series soon. Will include this ✅

**Comment by PatrykGala** [comment]
Hello @antoinebrl 
Thank you for your merge request. Upon review, we've identified some issues with the type annotations in our `neptune-client`:

- Many files are currently missing type annotations entirely.
- Some of the existing type annotations may be incorrect or outdated.

We have tested this commit and decided to merge it. However, we would like to gather feedback from you after the release.

Our goal is to bring full mypy support in `neptune-client` version 2.0.

File: pyproject.toml [hunk 1]
```diff
@@ -119,6 +119,7 @@ keywords = [
 ]
 packages = [
     { include = "neptune", from = "src" },
+    { include = "neptune/py.typed", from = "src" },
```
Comment (kgodlewski) [inline pyproject.toml:122]: This line can be safely removed, `include = "neptune"` will pull it in.

Comment (antoinebrl) [reply pyproject.toml:122]: @SiddhantSadangi can you revert your change?

Comment (SiddhantSadangi) [reply pyproject.toml]: I forgot I had made this change 🙈 
(need to get out of vacation mode)

**Review by PatrykGala — APPROVED** [review]

**Pull request merged**